ml-proto/host/builtins.ml

```diff
@@ -1,9 +1,50 @@
 open Source
+open Eval
+open Memory
+open Types
 
-let print vs =
+let print m vs =
   List.iter Print.print_value (List.map (fun v -> Some v) vs);
   None
 
+let rec stdout_write_inner at mem offset count i =
+  let load_result = Memory.load_extend mem (Int64.add offset i) Mem8 ZX Int32Type in
+
+  begin
+    match load_result with
+    | Values.Int32 byte ->
+      print_char (Char.chr (Int32.to_int byte))
+    | _ ->
+      ignore (Error.error at "load_extend returned wrong type")
```

Comment:
Would this be a use case where an `assert` would be appropriate? The type will only be wrong if `Memory.load_extend` produces a value with the wrong type.